test/tests.py

```diff
@@ -503,6 +513,32 @@ def verify_dmlc_messages(self, stderr, expected_msgs):
             self.expect_messages(
                 kind, actual_msgs.get(kind, []), expected_msgs.get(kind, []))
 
+    @staticmethod
+    def parse_depfile(f):
```

Comment:
Doesn't feel natural to have this as a method on DMLFileTestCase, the method is too specific. Its scope is highly specific and limited to dependency file tests. If you must extract it to a common place, then maybe a top-level function makes most sense. (But even better to not duplicate the DmlDep test)